### PR TITLE
Fix gcc9 warning in MuonAnalysis/MuonAssociators

### PR DESCRIPTION
**Pull request by mrodozov**
#### PR description:

Fixes warning in MuonAnalysis/MuonAssociators

#### PR validation:

Builds without warning

### CODE REVIEW

**Comment by mrodozov** [comment]
please test

File: MuonAnalysis/MuonAssociators/src/MatcherUsingTracksAlgorithm.cc [hunk 1]
```diff
@@ -182,6 +182,7 @@ bool MatcherUsingTracksAlgorithm::match(const reco::Candidate &c1,
         }
       }
     }
+      [[fallthrough]];
```
Comment (fabiocos) [inline MuonAnalysis/MuonAssociators/src/MatcherUsingTracksAlgorithm.cc]: @santocch @mrodozov a ```break``` would get the same effect here I would say, anyway it does not seem incorrect to me

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
merge

File: MuonAnalysis/MuonAssociators/src/MatcherUsingTracksAlgorithm.cc (hunk 1 above)
Comment (santocch) [reply MuonAnalysis/MuonAssociators/src/MatcherUsingTracksAlgorithm.cc]: I agree. The logic would be identical... I'm not sure which one would be better or what difference would a break introduce... probably none so let's keep this one.

**Comment by santocch** [comment]
+1

**Pull request merged**